src/senselab/audio/tasks/forced_alignment/forced_alignment.py

```diff
@@ -0,0 +1,747 @@
+
+    Args:
+        transcript (List[SingleSegment]): The list of transcription segments.
+        model_dictionary (Dict[str, int]): Dictionary for the alignment model.
+        model_lang (str): Language of the model.
+        print_progress (bool): Whether to print progress.
+        combined_progress (bool): Whether to combine progress percentage.
+
+    Returns:
+        List[SingleSegment]: The preprocessed transcription segments.
+    """
+    total_segments = len(transcript)
+
+    for sdx, segment in enumerate(transcript):
+        if print_progress:
+            base_progress = ((sdx + 1) / total_segments) * 100
+            percent_complete = (50 + base_progress / 2) if combined_progress else base_progress
+            print(f"Progress: {percent_complete:.2f}%...")
+
+        num_leading = len(segment["text"]) - len(segment["text"].lstrip())
+        num_trailing = len(segment["text"]) - len(segment["text"].rstrip())
+        text = segment["text"]
+
+        # Split into words
+        if model_lang not in LANGUAGES_WITHOUT_SPACES:
+            per_word = text.split(" ")
+        else:
+            per_word = [text]
+
+        clean_char, clean_cdx = [], []
+        for cdx, char in enumerate(text):
+            char_ = char.lower()
+            if model_lang not in LANGUAGES_WITHOUT_SPACES:
+                char_ = char_.replace(" ", "|")
+
+            if cdx < num_leading or cdx > len(text) - num_trailing - 1:
+                continue
+            elif char_ in model_dictionary.keys():
+                clean_char.append(char_)
+                clean_cdx.append(cdx)
+
+        clean_wdx = []
+        for wdx, wrd in enumerate(per_word):
+            if any(c in model_dictionary.keys() for c in wrd):
+                clean_wdx.append(wdx)
+
+        punkt_param = PunktParameters()
+        punkt_param.abbrev_types = set(PUNKT_ABBREVIATIONS)
+        sentence_splitter = PunktSentenceTokenizer(punkt_param)
+        sentence_spans = list(sentence_splitter.span_tokenize(text))
+
+        segment["clean_char"] = clean_char
+        segment["clean_cdx"] = clean_cdx
+        segment["clean_wdx"] = clean_wdx
+        segment["sentence_spans"] = sentence_spans
+
+    return transcript
+
+
+def _can_align_segment(
+    segment: SingleSegment, model_dictionary: Dict[str, int], t1: float, max_duration: float
+) -> bool:
+    """Checks if a segment can be aligned.
+
+    Args:
+        segment (SingleSegment): The segment to check.
+        model_dictionary (Dict[str, int]): Dictionary for character indices.
+        t1 (float): Start time of the segment.
+        max_duration (float): Maximum duration of the audio.
+
+    Returns:
+        bool: True if the segment can be aligned, False otherwise.
+    """
+    if segment["clean_char"] is None or len(segment["clean_char"]) == 0:
+        return False
+    if t1 >= max_duration:
+        return False
+    return True
+
+
+def _prepare_waveform_segment(
```

Comment:
we do have a method doing exactly the same in our preprocessing module...

Reply:
This one is a little different because it pads the audio segment, but I updated the preprocessing module and replaced this funciton.